Fix: The values of StatusCodes shown on Swagger UI of mentorship_relation API changed  to be in number format instead of Enum Representations.

Aaishpra opened the pull request:
### Description
The values of  HTTPStatus codes shown on Swagger UI of mentorship_relation API changed to be in number format instead of Enum Representations. 
<!--- Include a summary of the change and relevant motivation/context. List any dependencies that are required for this change. --->

Fixes #939

### Type of Change:

<!--- **Delete irrelevant options.** --->

- Code

- User Interface


**Code/Quality Assurance Only**
- Bug fix (non-breaking change which fixes an issue)

### How Has This Been Tested?
Go to ``` mentorship_relation```
Click on ```Post/mentorship_relation/send_request```(or any other task)
See Results

![image](https://user-images.githubusercontent.com/66299533/103148707-a5ef9b80-4788-11eb-9187-9dc84c226778.png)

<!-- Describe the tests you ran to verify your changes. Provide instructions or GIFs so we can reproduce. List any relevant details for your test. -->


### Checklist:

<!-- **Delete irrelevant options.** -->

- [x] My PR follows the style guidelines of this project
- [x] I have performed a self-review of my own code or materials


**Code/Quality Assurance Only**

- [x] My changes generate no new warnings 
- [x] My PR currently breaks something (fix or feature that would cause existing functionality to not work as expected)
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been published in downstream modules

Aaishpra committed:
Enum representation of status codes changed to real value

Aaishpra commented:
@isabelcosta @devkapilbansal can anyone of you review it?
Also, I don't understand why these tests are failing.

devkapilbansal commented:
@Aaishpra these tests failed due to some dependencies issues which will soon be resolved

devkapilbansal committed:
fix: upgrade Flask_JWT_Extended to ensure all tests passed (#945)

isabelcosta commented:
@Aaishpra can you please update your branch to have the latest changes? These will fix the tests :) hopefully 🙏 
cc @devkapilbansal

Aaishpra committed:
Update Fork

Aaishpra commented:
> @Aaishpra can you please update your branch to have the latest changes? These will fix the tests :) hopefully pray
> cc @devkapilbansal

Done, but still some tests are failing.

isabelcosta commented:
@Aaishpra the tests are passing fine. don't worry about **codecov/patch** check, this is a common fail, we are not sure why yet. Regarding code formatting, if you run `black app/api/resources/mentorship_relation.py` this will reformat code according to [black](https://github.com/psf/black) tool standards

Aaishpra commented:
> @Aaishpra the tests are passing fine. don't worry about **codecov/patch** check, this is a common fail, we are not sure why yet. Regarding code formatting, if you run `black app/api/resources/mentorship_relation.py` this will reformat code according to [black](https://github.com/psf/black) tool standards

Ahh thanks, for telling.
Another thing can i get assigned to any one of the new issues which i opened😁
#946 #947

isabelcosta commented:
> > @Aaishpra the tests are passing fine. don't worry about **codecov/patch** check, this is a common fail, we are not sure why yet. Regarding code formatting, if you run `black app/api/resources/mentorship_relation.py` this will reformat code according to [black](https://github.com/psf/black) tool standards
> 
> Ahh thanks, for telling.
> Another thing can i get assigned to any one of the new issues which I opened😁

I would prefer if you could fix this issue first, then we can leave another available, and once you finished your change requests here you can move on to the next one, and leave another one for a first-time contributor. I will label them accordingly. So if by the time you finish your part in this PR, you can move to another one. Just ask there to be assigned to it :) @Aaishpra

Aaishpra commented:
> > > @Aaishpra the tests are passing fine. don't worry about **codecov/patch** check, this is a common fail, we are not sure why yet. Regarding code formatting, if you run `black app/api/resources/mentorship_relation.py` this will reformat code according to [black](https://github.com/psf/black) tool standards
> > 
> > Ahh thanks, for telling.
> > Another thing can i get assigned to any one of the new issues which I opened😁
> 
> I would prefer if you could fix this issue first, then we can leave another available, and once you finished your change requests here you can move on to the next one, and leave another one for a first-time contributor. I will label them accordingly. So if by the time you finish your part in this PR, you can move to another one. Just ask there to be assigned to it :) @Aaishpra 

Sure thing 👍

diananova committed:
feat: return user registration_time field in GET /users (#943)

tichnas committed:
docs: refresh endpoint shows refresh token instead of access token (#936)

surbhigoel191 committed:
fix: broken links on Swagger UI (#951)

devkapilbansal commented:
> @Aaishpra the tests are passing fine. don't worry about **codecov/patch** check, this is a common fail, we are not sure why yet. Regarding code formatting, if you run `black app/api/resources/mentorship_relation.py` this will reformat code according to [black](https://github.com/psf/black) tool standards

@isabelcosta may be I know the reason why codecov/patch failed.

https://github.com/anitab-org/mentorship-backend/blob/9618e2c697f5f4c0bdc851a887e4a2a47a39122a/tests/mentorship_relation/test_api_send_request.py#L89

Here, hard coded status code is used, probable fixing these in tests can increase codecov/patch coverage.
Also, most of the tests are not covered in the tests file, like every possible error should be there in the test. We can work on that too

Aaishpra committed:
Code formatted with black

Aaishpra committed:
i "merge patch-1"
:wq

Merge branch 'patch-1' of https://github.com/Aaishpra/mentorship-backend into patch-1

Aaishpra commented:
> @Aaishpra the tests are passing fine. don't worry about **codecov/patch** check, this is a common fail, we are not sure why yet. Regarding code formatting, if you run `black app/api/resources/mentorship_relation.py` this will reformat code according to [black](https://github.com/psf/black) tool standards

@isabelcosta Done, please review and suggest any other change if required.

devkapilbansal commented on app/api/resources/mentorship_relation.py:
```diff
@@ -92,12 +93,12 @@ def post(cls):
 
         if is_valid != {}:
-            return is_valid, HTTPStatus.BAD_REQUEST
+            return is_valid, HTTPStatus.BAD_REQUEST.value
```
```suggestion
            return is_valid, HTTPStatus.BAD_REQUEST
```

devkapilbansal reviewed (changes requested):
@Aaishpra remove `.value` from return statements as asked by @isabelcosta [here](https://github.com/anitab-org/mentorship-backend/pull/952#discussion_r549875920)

Aaishpra commented on app/api/resources/mentorship_relation.py:
```diff
@@ -92,12 +93,12 @@ def post(cls):
 
         # if the mentorship relation creation failed dont send email and return
-        if response[1] != HTTPStatus.CREATED:
+        if response[1] != HTTPStatus.CREATED.value:
```
shall I remove .value from here too.
@devkapilbansal

devkapilbansal replied:
No, it looks fine

Aaishpra replied:
Done @devkapilbansal

Aaishpra committed:
removed .value from return statements

aakankshaagr committed:
fix: show HTTP status codes real value for Admin API (#956)

surbhigoel191 committed:
fix: use http code values instead of expressions for User API (#953)

Aaishpra commented:
@devkapilbansal @isabelcosta  please review the changes.

vj-codes reviewed (changes requested):
@Aaishpra can you please remove the change made in` requirements.txt` as it does not deal with the issue, thanks!

Aaishpra commented:
> @Aaishpra can you please remove the change made in` requirements.txt` as it does not deal with the issue, thanks!

I had to update my fork, because of some failing check that's why there is a change reflected in 'requirement.txt'.

NenadPantelic committed:
fix: change response status codes for invalid inputs (409 to 400) (#959)

* change response status codes for invalid inputs (409 to 400)
* Add tests to check invalid input cases.

battuAshita committed:
docs: update environment variable declarations (#974)

Updated the environment variable declarations by removing the whitespace character as it shows an error.
This resolves the issue #965.

gaurivn committed:
fix: refactor HTTP error codes on Task creation API (#734)

* fix: refactor HTTP error codes POST /mentorship_relation/{relation_id}/task/{task_id}/comment

devkapilbansal commented:
> > @Aaishpra can you please remove the change made in` requirements.txt` as it does not deal with the issue, thanks!
> 
> I had to update my fork, because of some failing check that's why there is a change reflected in 'requirement.txt'.

@Aaishpra that change in requirements may be there because tip of your branch is behind. You should update it with latest develop branch.

Aaishpra commented:
@devkapilbansal I don't think the change in requirement.txt will cause any problem after this PR is merged as the change is already there in the develop branch of this repo, but I think updating my fork again will cause only extra commits in the PR which ain't required. 
If you still think updating my fork is required just ping me again I will update it.

devkapilbansal commented:
> @devkapilbansal I don't think the change in requirement.txt will cause any problem after this PR is merged as the change is already there in the develop branch of this repo, but I think updating my fork again will cause only extra commits in the PR which ain't required.
> If you still think updating my fork is required just ping me again I will update it.

It may not cause extra commits. You can soft reset your changes to current develop branch.
Something like:- 
```
git remote add upstream https://github.com/anitab-org/mentorship-backend.git
git upstream fetch develop
git reset --soft upstream/develop
git commit -m "Your message here"
```
Hope it helps 😊

Aaishpra committed:
fork update

Aaishpra committed:
fork update
Merge branch 'patch-1' of https://github.com/Aaishpra/mentorship-backend into patch-1

Aaishpra commented:
@devkapilbansal thanks for the help although I think messed somethings which have caused all this mess.

That's why closing this PR.
